### DISCUSSION TITLE
Beatmap saved without hitobjects

### DISCUSSION THREAD

**Discussion by get2day** [ppy/osu#24020] [Ideas] [answered]
Today, I was mapping an beatmap in osu!lazer (2023.621.0) and suddenly the beatmap was corrupted after saving, lost all data for one difficulty and reverted data for an another difficulty, which costs hours of work.

~~To fix this issue, auto-backups can save the data lost if the beatmap has been corrupted.
Auto-backups are also useful to revert the beatmap back to the old version when a mapper is unsatisfied with the beatmap.~~

**Comment by peppy** [comment]
Could you please attach the corrupted file?

**Reply by get2day** [reply]
[Corrupted Beatmap.zip](https://github.com/ppy/osu/files/11850407/Corrupted.Beatmap.zip)
The corrupted file is in the zip.

**Reply by get2day** [reply]
After some Further Research,
In Stable, the beatmap seems completely fine.
However, in Lazer, the beatmap's data somehow cannot be readed.

**Reply by peppy** [reply]
This doesn't look corrupted, but does look missing any hitobjects. I'm wondering if you somehow triggered a save during loading or exit when hitobjects weren't available..

**Reply by get2day** [reply]
I think what're you saying is correct, I somehow triggered a save during exit when hitobjects weren't available.

When I imported the beatmap to stable, the difficulty "Perfect Premonition" has objects went missing.

~~Anyways, I think Auto-backups are essential when something like this happens.~~

**Reply by peppy** [reply, marked as answer]
The hope is that we could avoid this kind of thing in the first place rather than a convoluted backup restoration progress when other users inevitably hit the same issue.

**Reply by peppy** [reply]
Looking through the code, I cannot feasibly see how this can happen.

I don't suppose you kept log files from when it happened?

**Reply by get2day** [reply]
I haven't kept the log files from when it happened, but I will check if I could find something in the log files later.

**Reply by peppy** [reply]
I've opened a tracking issue for this https://github.com/ppy/osu/issues/24031.
